test/model/sequence_result_test.rb

```diff
@@ -3,11 +3,310 @@
+    test_result1.required = true
+    test_result1.result = Inferno::ResultStatuses::PASS
+    sequence_result.test_results << test_result1
+
+    test_result2 = Inferno::Models::TestResult.new
+    test_result2.required = true
+    test_result2.result = Inferno::ResultStatuses::TODO
+    sequence_result.test_results << test_result2
+
+    sequence_result.update_result_counts
+
+    assert_equal Inferno::ResultStatuses::PASS, sequence_result.result
+  end
+
+  def test_sequence_result_result_pass_even_with_omit
+    sequence_result = Inferno::Models::SequenceResult.new
+    sequence_result.test_results = []
+
+    test_result1 = Inferno::Models::TestResult.new
+    test_result1.required = true
+    test_result1.result = Inferno::ResultStatuses::PASS
+    sequence_result.test_results << test_result1
+
+    test_result2 = Inferno::Models::TestResult.new
+    test_result2.required = true
+    test_result2.result = Inferno::ResultStatuses::OMIT
+    sequence_result.test_results << test_result2
+
+    sequence_result.update_result_counts
+
+    assert_equal Inferno::ResultStatuses::PASS, sequence_result.result
+  end
+
+  def test_sequence_result_result_pass_when_skip_test_is_optional
+    sequence_result = Inferno::Models::SequenceResult.new
+    sequence_result.test_results = []
+
+    test_result1 = Inferno::Models::TestResult.new
+    test_result1.required = true
+    test_result1.result = Inferno::ResultStatuses::PASS
+    sequence_result.test_results << test_result1
+
+    test_result2 = Inferno::Models::TestResult.new
+    test_result2.required = false
+    test_result2.result = Inferno::ResultStatuses::SKIP
+    sequence_result.test_results << test_result2
+
+    sequence_result.update_result_counts
+
+    assert_equal Inferno::ResultStatuses::PASS, sequence_result.result
+  end
+
+  def test_sequence_result_result_skip_when_skip_test_is_required
+    sequence_result = Inferno::Models::SequenceResult.new
+
+    test_result1 = Inferno::Models::TestResult.new
+    test_result1.required = true
+    test_result1.result = Inferno::ResultStatuses::PASS
+    sequence_result.test_results << test_result1
+
+    test_result2 = Inferno::Models::TestResult.new
+    test_result2.required = true
+    test_result2.result = Inferno::ResultStatuses::SKIP
+    sequence_result.test_results << test_result2
+
+    sequence_result.update_result_counts
+
+    assert_equal Inferno::ResultStatuses::SKIP, sequence_result.result
+  end
+
+  def test_sequence_result_result_required_skip_when_no_pass_tests
+    sequence_result = Inferno::Models::SequenceResult.new
+
+    test_result1 = Inferno::Models::TestResult.new
+    test_result1.required = true
+    test_result1.result = Inferno::ResultStatuses::SKIP
+    sequence_result.test_results << test_result1
+
+    sequence_result.update_result_counts
+
+    assert_equal Inferno::ResultStatuses::SKIP, sequence_result.result
+  end
+
+  def test_sequence_result_result_optional_skip_when_no_pass_tests
```

Comment:
In this test, sequence_result does not result "optional_skip". Should change to a better name?

Reply:
changed name to test_sequence_result_pass_when_only_1_optional_skip_test